code/game/gamemodes/game_mode.dm

```diff
@@ -603,3 +606,120 @@
 		SSticker.news_report = STATION_EVACUATED
 		if(SSshuttle.emergency.is_hijacked())
 			SSticker.news_report = SHUTTLE_HIJACK
+
+///////////////////////////////////
+//Keeps track of all living heads//
+///////////////////////////////////
+/datum/game_mode/proc/get_living_by_department(var/department)
+	. = list()
+	for(var/mob/living/carbon/human/player in GLOB.mob_list)
+		if(player.stat != DEAD && player.mind && (player.mind.assigned_role in department))
+			. |= player.mind
+
+
+////////////////////////////
+//Keeps track of all heads//
+////////////////////////////
+/datum/game_mode/proc/get_all_by_department(var/department)
+	. = list()
+	for(var/mob/player in GLOB.mob_list)
+		if(player.mind && (player.mind.assigned_role in department))
+			. |= player.mind
+
+/////////////////////////////////////////////
+//Keeps track of all living silicon members//
+/////////////////////////////////////////////
+/datum/game_mode/proc/get_living_silicon()
+	. = list()
+	for(var/mob/living/silicon/player in GLOB.mob_list)
+		if(player.stat != DEAD && player.mind && (player.mind.assigned_role in GLOB.nonhuman_positions))
+			. |= player.mind
+
+///////////////////////////////////////
+//Keeps track of all silicon members //
+///////////////////////////////////////
+/datum/game_mode/proc/get_all_silicon()
+	. = list()
+	for(var/mob/living/silicon/player in GLOB.mob_list)
+		if(player.mind && (player.mind.assigned_role in GLOB.nonhuman_positions))
+			. |= player.mind
+
+/datum/game_mode/proc/generate_credit_text()
+		var/list/round_credits = list()
+		var/len_before_addition
+
+		// HEADS OF STAFF
+		round_credits += "<center><h1>The Glorious Command Staff:</h1>"
+		len_before_addition = round_credits.len
+		for(var/datum/mind/current in SSticker.mode.get_all_by_department(GLOB.command_positions))
+				round_credits += "<center><h2>[current.name] as the [current.assigned_role]</h2>"
+		if(round_credits.len == len_before_addition)
+				round_credits += list("<center><h2>A serious bureaucratic error has occurred!</h2>", "<center><h2>No one was in charge of the crew!</h2>")
+		round_credits += "<br>"
+
+		// SILICONS
+		round_credits += "<center><h1>The Silicon \"Intelligences\":</h1>"
+		len_before_addition = round_credits.len
+		for(var/datum/mind/current in SSticker.mode.get_all_silicon())
+				round_credits += "<center><h2>[current.name] as the [current.assigned_role]</h2>"
+		if(round_credits.len == len_before_addition)
+				round_credits += list("<center><h2>[station_name()] had no silicon helpers!</h2>", "<center><h2>Not a single door was opened today!</h2>")

```

Comment:
This is untrue; humans can still open doors on their own.

/nitpick

Reply:
That's the joke.